xtable-core/src/main/java/org/apache/xtable/iceberg/IcebergConversionSource.java

```diff
@@ -120,10 +120,18 @@ public InternalTable getTable(Snapshot snapshot) {
         irPartitionFields.size() > 0
             ? DataLayoutStrategy.HIVE_STYLE_PARTITION
             : DataLayoutStrategy.FLAT;
+    // When the table name is not explicitly specified, Iceberg assumes the table is HDFS-based,
+    // treating the table name as the location in HDFS. This assumption can lead to mismatches
+    // during metadata conversion. To mitigate this issue, we rely on the table name provided in the
+    // source configuration of the conversation so target matches the user's expectations.
+    // See https://github.com/apache/incubator-xtable/issues/494
     return InternalTable.builder()
         .tableFormat(TableFormat.ICEBERG)
         .basePath(iceTable.location())
-        .name(iceTable.name())
+        .name(
+            iceTable.name().contains(iceTable.location())
+                ? sourceTableConfig.getName()
+                : iceTable.name())
```

Comment:
Could you please confirm if existing tests cases cover both the paths?

Reply:
Added new test with catalog-config

Reply:
@ashvina I confirmed that both branches are hit executed in our tests now. Is there anything else blocking this bug fix?